app/src/main/java/com/antosik/benchproject/app/common/binding/recyclerView/RecyclerViewBindingAdapters.kt

```diff
@@ -6,6 +6,8 @@ import com.antosik.benchproject.app.common.recyclerView.RecyclerBaseAdapter
 
 @Suppress("UNCHECKED_CAST")
 @BindingAdapter("adapterData")
-fun RecyclerView.updateAdapterData(data: List<Any>) {
-    (adapter as RecyclerBaseAdapter<Any>).uploadData(data)
+fun RecyclerView.updateAdapterData(data: Any?) {
+    if (data != null) {
+        (adapter as RecyclerBaseAdapter<Any>).uploadData(data as List<Any>)
```

Comment:
Why there is a casting here instead of just declaring `data` as `List<Any>`?

Reply:
IMO done 😅

Reply:
I created an issue for rethink about extra `LiveData` for data from `UIState` #165